### PR TITLE
[Move] Finish implementation of Glaive Rush

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes?
The secondary effects of [Glaive Rush](https://bulbapedia.bulbagarden.net/wiki/Glaive_Rush_(move)) are implemented. A nearby unused class was also removed.

## Why am I doing these changes?
The move is not fully implemented yet.

## What did change?
The secondary effects of Glaive Rush were implemented via adding the tags `BattlerTag.ALWAYS_GET_HIT` and `BattlerTag.RECEIVE_DOUBLE_DAMAGE`, and the necessary classes to accompany them. Unit tests were also added.

## How to test the changes?
`npm run test glaive_rush`

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -4142,17 +4142,29 @@ export class IgnoreAccuracyAttr extends AddBattlerTagAttr {
   }
 }
 
-export class AlwaysCritsAttr extends AddBattlerTagAttr {
+export class AlwaysGetHitAttr extends AddBattlerTagAttr {
   constructor() {
-    super(BattlerTagType.ALWAYS_CRIT, true, false, 2);
+    super(BattlerTagType.ALWAYS_GET_HIT, true, false);
```
Comment (innerthunder) [inline src/data/move.ts:4147]:
```suggestion
    super(BattlerTagType.ALWAYS_GET_HIT, true, false, 0, 0, true);
```
This sets `lastHitOnly` to `true` for this attribute. PRE_MOVE battler tags tend to have weird overlapping issues when they come from multi-hit moves, and this change avoids those issues.

File: src/data/move.ts [hunk 2]
```diff
@@ -4142,17 +4142,29 @@ export class IgnoreAccuracyAttr extends AddBattlerTagAttr {
...
     }
 
-    user.scene.queueMessage(getPokemonMessage(user, ` took aim\nat ${target.name}!`));
+    return true;
+  }
+}
+
+export class ReceiveDoubleDamageAttr extends AddBattlerTagAttr {
+  constructor() {
+    super(BattlerTagType.RECEIVE_DOUBLE_DAMAGE, true, false);
```
Comment (innerthunder) [inline src/data/move.ts:4161]:
```suggestion
    super(BattlerTagType.RECEIVE_DOUBLE_DAMAGE, true, false, 0, 0, true);
```
Same as the other attribute.

File: src/test/moves/glaive_rush.test.ts
Comment (innerthunder) [inline src/test/moves/glaive_rush.test.ts:1]: I'd recommend adding a test where the Glaive Rush user has Multi-Lens to make sure tag overlapping isn't an issue.

**Comment by DayKev** [comment]
Closing due to this becoming a duplicate PR from the new contribution workflow.

**Comment by DayKev** [comment]
Nvm apparently I misunderstood what happened.

**Comment by DayKev** [comment]
Alright, should be fully updated now.

**Pull request merged**